onnxruntime/core/framework/execution_provider.cc

```diff
@@ -51,14 +51,25 @@ common::Status IExecutionProvider::OnRunEnd() { return Status::OK(); }
 
 common::Status IExecutionProvider::OnSessionInitializationEnd() { return Status::OK(); }
 
+// Update allocator in the provider if already present; ignore if not.
+void IExecutionProvider::UpdateAllocator(AllocatorPtr allocator) {
+  const auto& info = allocator->Info();
+  auto ite = mem_info_set_.find(info);
+  if (ite != mem_info_set_.end()) {
+    const int key = MakeKey(info.id, info.mem_type);
+    allocators_[key] = allocator;
```

Comment:
Is there an inconsistency here between what fields in OrtMemoryInfo are used as the key in mem_info_set_ vs allocators_? Any chance that would lead to unexpected behavior?

Reply:
Should not. Not sure why the lookup was based on id and mem type only before. I retained that while ensuring uniqueness.